sources/libs/virtio/block_device.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <brutal/base.h>
+#include <virtio/device.h>
+#include <virtio/virtqueue.h>
+
+// All the features of a Virtio Block Device
+typedef enum
+{
+    VIRTIO_BLK_F_SIZE_MAX = 1 << 1,
+    VIRTIO_BLK_F_SEG_MAX = 1 << 2,
+    VIRTIO_BLK_F_GEOMETRY = 1 << 4,
+    VIRTIO_BLK_F_RO = 1 << 5,
+    VIRTIO_BLK_F_BLK_SIZE = 1 << 6,
+    VIRTIO_BLK_F_FLUSH = 1 << 9,
+    VIRTIO_BLK_F_TOPOLOGY = 1 << 10,
+    VIRTIO_BLK_F_CONFIG_WCE = 1 << 11,
+    VIRTIO_BLK_F_DISCARD = 1 << 13,
+    VIRTIO_BLK_F_WRITE_ZEROES = 1 << 14,
+} VirtioBlkDeviceFeatures;
+
+// A Virtio Block Device config
+typedef struct PACKED
+{
+    le_uint64_t capacity;
+    le_uint32_t size_max;
+    le_uint32_t seg_max;
+
+    struct PACKED virtio_blk_geometry
+    {
+        le_uint16_t cylinders;
+        uint8_t heads;
+        uint8_t sectors;
+    } geometry;
+
+    le_uint32_t block_size;
+
+    struct PACKED virtio_blk_topology
```

Comment:
maybe you should separates these structs or don't put these in struct like:

```c


    le_uint32_t seg_max;

    struct PACKED virtio_blk_geometry
    {
        le_uint16_t cylinders;
        uint8_t heads;
        uint8_t sectors;
    } geometry;

    le_uint32_t block_size;

```

to

```c

    le_uint32_t seg_max;
    le_uint16_t cylinders;
    uint8_t heads;
    uint8_t sectors;

    le_uint32_t block_size;

```
or:
```c


    le_uint32_t seg_max;
   
    VirtioBlkGeometry geometry;

    le_uint32_t block_size;

```